libsolidity/interface/CompilerStack.cpp

```diff
@@ -711,9 +711,15 @@ void CompilerStack::compileContract(
 	for (auto const* dependency: _contract.annotation().contractDependencies)
 		compileContract(*dependency, _compiledContracts);
 
-	shared_ptr<Compiler> compiler = make_shared<Compiler>(m_evmVersion, m_optimize, m_optimizeRuns);
 	Contract& compiledContract = m_contracts.at(_contract.fullyQualifiedName());
+
+	shared_ptr<Compiler> compiler = make_shared<Compiler>(m_evmVersion, m_optimize, m_optimizeRuns);
```

Comment:
Perhaps it is better to not introduce this local variable at all? Same goes for `metadata` below.

Reply:
`metadata` is used twice: here and for hashing. Though it could use `compiledContract.metadata`.

Reply:
`compiler` is used at least 3 times, it probably makes sense to have the local variable for readability.